cfgov/unprocessed/js/apps/financial-well-being/fwb-questions.js

```diff
@@ -1,83 +1,169 @@
-/* eslint no-global-assign: "off" */
 'use strict';
 
-var Analytics = require( '../../modules/Analytics' );
+const Analytics = require( '../../modules/Analytics' );
 
+/**
+ * Initialize the questionnaire
+ */
 function init() {
+  const questionsDom = document.querySelectorAll( '#quiz-form fieldset' );
+  const radiosDom = document.querySelectorAll( '#quiz-form [type="radio"]' );
+  const submitDom = document.querySelector( '#submit-quiz' );
+  const questionStates = {};
+
+  /**
+   * Checks the selection status of a group of radio inputs,
+   * belonging to a parent fieldset
+   * @param {HTMLNode} questionRadios An array of DOM elements.
+   * @returns {boolean} The status of the group of inputs
+   */
+  function checkQuestionState( questionRadios ) {
+    let radioIsChecked = false;
+
+    for ( let i = 0, l = questionRadios.length; i < l; i++ ) {
+      if ( questionRadios[i].checked ) {
+        radioIsChecked = true;
+      }
+    }
 
-  var totalQuestions = 12;
-  var quizComplete = false;
-  var questionValues = {};
-  var radioButtons = document.querySelectorAll( '[type="radio"]' );
-  var submitButton = document.querySelector( '#submit-quiz' );
+    return radioIsChecked;
+  }
 
-  function allQuestionsCompleted() {
-    return Object.keys( questionValues ).length === totalQuestions;
+  /**
+   * Checks the status of each fieldset to determine if
+   * the user has completed the form
+   * @returns {boolean} The completion status of the form
+   */
+  function checkFormCompletion() {
+    let formCompleted = true;
+
+    Object.keys( questionStates ).forEach( function( key ) {
```

Comment:
trivial: You could use double arrow here.

Reply:
Oh good call.